src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BuildTest.cs

```diff
@@ -328,17 +328,29 @@ public void CheckAssemblyCounts ([Values (true, false)] bool isRelease, [Values
 		// DotNet fails, see https://github.com/dotnet/runtime/issues/65484
 		// Enable the commented out signature (and AOT) once the above is fixed
 		[Test]
-		public void SmokeTestBuildWithSpecialCharacters ([Values (false, true)] bool forms, [Values (false /*, true*/)] bool aot)
+		public void SmokeTestBuildWithSpecialCharacters ([Values (false, true)] bool forms, [Values (false, true)] bool aot, [Values] AndroidRuntime runtime)
 		{
+			if (IgnoreUnsupportedConfiguration (runtime, aot: aot, release: true)) {
+				return;
+			} else if (!aot && runtime == AndroidRuntime.NativeAOT) {
+				// Just saving time, aot && !aot would be indentical tests with NativeAOT runtime
+				Assert.Ignore ("NativeAOT always uses AOT, obviously");
+				return;
+			}
+
 			var testName = "テスト";
 
 			var rootPath = Path.Combine (Root, "temp", TestName);
 			var proj = forms ?
 				new XamarinFormsAndroidApplicationProject () :
 				new XamarinAndroidApplicationProject ();
+			proj.SetRuntime (runtime);
 			proj.ProjectName = testName;
 			proj.IsRelease = true;
-			proj.AotAssemblies = aot;
+
+			if (runtime != AndroidRuntime.NativeAOT) {
+				proj.AotAssemblies = aot;
+			}
```

Comment:
This property doesn't do anything on CoreCLR either, so maybe:

```suggestion
			if (runtime == AndroidRuntime.MonoVM) {
				proj.AotAssemblies = aot;
			}
```